src/lazy/text/raw/sequence.rs

```diff
@@ -0,0 +1,126 @@
+    fn from_value(value: LazyRawTextValue<'data>) -> Self {
+        LazyRawTextSequence { value }
+    }
+}
+
+impl<'data> LazyRawSequence<'data, TextEncoding> for LazyRawTextSequence<'data> {
+    type Iterator = RawTextSequenceIterator<'data>;
+
+    fn annotations(&self) -> <TextEncoding as LazyDecoder<'data>>::AnnotationsIterator {
+        todo!("lazy sequence annotations")
+    }
+
+    fn ion_type(&self) -> IonType {
+        self.value.ion_type()
+    }
+
+    fn iter(&self) -> Self::Iterator {
+        LazyRawTextSequence::iter(self)
+    }
+
+    fn as_value(&self) -> &LazyRawTextValue<'data> {
+        &self.value
+    }
+}
+
+impl<'a, 'data> IntoIterator for &'a LazyRawTextSequence<'data> {
+    type Item = IonResult<LazyRawTextValue<'data>>;
+    type IntoIter = RawTextSequenceIterator<'data>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl<'a> Debug for LazyRawTextSequence<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self.value.encoded_value.ion_type() {
+            IonType::SExp => {
+                write!(f, "(")?;
+                for value in self {
+                    write!(
+                        f,
+                        "{:?} ",
+                        value
+                            .map_err(|_| fmt::Error)?
+                            .read()
+                            .map_err(|_| fmt::Error)?
+                    )?;
+                }
+                write!(f, ")").unwrap();
+            }
+            IonType::List => {
+                write!(f, "[")?;
+                for value in self {
+                    write!(
+                        f,
+                        "{:?},",
+                        value
+                            .map_err(|_| fmt::Error)?
+                            .read()
+                            .map_err(|_| fmt::Error)?
+                    )?;
+                }
+                write!(f, "]").unwrap();
+            }
+            _ => unreachable!("LazyRawSequence is only created for list and sexp"),
+        }
+
+        Ok(())
+    }
+}
+
+pub struct RawTextSequenceIterator<'data> {
+    input: TextBufferView<'data>,
+}
+
+impl<'data> RawTextSequenceIterator<'data> {
+    pub(crate) fn new(input: TextBufferView<'data>) -> RawTextSequenceIterator<'data> {
+        RawTextSequenceIterator { input }
+    }
+}
+
+impl<'data> Iterator for RawTextSequenceIterator<'data> {
+    type Item = IonResult<LazyRawTextValue<'data>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.input.match_list_value() {
+            Ok((remaining, Some(value))) => {
+                self.input = remaining;
+                Some(Ok(value))
+            }
+            Ok((_remaining, None)) => None,
+            Err(e) => e
+                .with_context("reading the next list value", self.input)
+                .transpose(),
```

Comment:
🗺️ This call to `transpose()` is the `ToIteratorOutput` trait's extension method.